src/app/features/search-bar/search-bar.component.ts

```diff
@@ -0,0 +1,250 @@
+        if (ev.key === 'Escape') {
+          this.blurred.emit();
+        } else if (ev.key === '1' && ev.ctrlKey) {
+          this.switchTaskSource();
+          ev.preventDefault();
+        } else if (
+          ev.key === 'Enter' &&
+          (!this.taskSuggestionsCtrl.value ||
+            typeof this.taskSuggestionsCtrl.value === 'string')
+        ) {
+          this._shakeSearchForm();
+          ev.preventDefault();
+        }
+      });
+    });
+  }
+
+  private _triggerFilter(): void {
+    this.taskSuggestionsCtrl.setValue(this.inputEl.nativeElement.value);
+  }
+
+  private _setUpFilter(): void {
+    this.filteredIssueSuggestions$ = this.taskSuggestionsCtrl.valueChanges.pipe(
+      debounceTime(100),
+      startWith(''),
+      tap(() => this.isLoading$.next(true)),
+      map((searchTerm) => this._filter(searchTerm)),
+      tap(() => this.isLoading$.next(false)),
+    );
+  }
+
+  private _getLocation(item: SearchItem): string {
+    const tasksOrWorklog = !this.isArchivedTasks ? 'tasks' : 'worklog';
+    if (item.projectId) {
+      return `/project/${item.projectId}/${tasksOrWorklog}`;
+    } else if (item.tagIds.includes(TODAY_TAG.id)) {
+      return `/tag/TODAY/${tasksOrWorklog}`;
+    } else if (item.tagIds[0]) {
+      return `/tag/${item.tagIds[0]}/${tasksOrWorklog}`;
+    } else {
+      console.warn("Couldn't find task location");
+      return '';
+    }
+  }
+
+  private _getArchivedDate(item: SearchItem): string {
+    if (Object.keys(item.timeSpentOnDay)[0]) return Object.keys(item.timeSpentOnDay)[0];
+    if (item.createdOn) return getWorklogStr(item.createdOn);
+    const parentTask = this._tasks.find((t) => t.id === item.id);
+    if (parentTask?.created) return getWorklogStr(parentTask.created);
+    return '';
+  }
+
+  private _getCtxForTaskSuggestion(task: Task): Tag | Project {
+    if (task.projectId) {
+      return this._projects.find((project) => project.id === task.projectId) as Project;
+    } else {
+      if (!task.tagIds[0]) {
+        throw new Error('No first tag');
+      }
+      return this._tags.find((tag) => tag.id === task.tagIds[0]) as Tag;
+    }
+  }
+
+  private _isInBacklog(item: SearchItem): boolean {
+    if (!item.projectId) return false;
+    const project = this._projects.find((p) => p.id === item.projectId);
+    return project ? project.backlogTaskIds.includes(item.id) : false;
+  }
+
+  private _filter(searchTerm: string): SearchItem[] {
+    let result = this._searchableItems.filter(
+      (task) =>
+        task.title.includes(searchTerm.toLowerCase()) ||
+        task.taskNotes.includes(searchTerm.toLowerCase()),
+    );
+
+    if (result.length > 200) {
+      this.tooManyResults = true;
+      result = result.slice(0, 200);
+    } else {
+      this.tooManyResults = false;
+    }
+
+    return result;
+  }
+
+  private _shakeSearchForm(): void {
+    this.searchForm.nativeElement.classList.toggle('shake-form');
```

Comment:
Oh nice! I wasn't aware that `classList.toggle` exists :)

Reply:
had to google it myself :)